platform/android/MapboxGLAndroidSDKTestApp/src/androidTest/java/com/mapbox/mapboxsdk/testapp/maps/widgets/CompassViewTest.java

```diff
@@ -40,19 +38,21 @@ public void testDefault() {
   }
 
   @Test
+  @Ignore
   public void testVisible() {
```

Comment:
@tobrun IMO the way you should treat test code is not the same as the way we are accustomed to write production code.

> A method selector is always necessary to refer to the method from other places. Not necessarily a name, but some kind of selector is needed. That’s not true for test names, and to a degree it’s interesting that the default of having test names comes from the implementation artifact of tests usually being represented as methods. Would we still have test names in all cases if we didn’t start designing a test framework based around a language that cripples usage of abstractions other than methods for executable code? –Ola Bini

Ultimately a test name is a description of _why_ that test exists (intent). A _required_ description (because of JUnit). I see test names as _glorified comments_ and I tend to write them to be self-explanatory without a name.

Most of my thoughts are based on the awesome book [Working Effectively With Unit Tests](http://wewut.com/). I really like the techniques described on it favoring DAMP (Descriptive And Maintainable Procedures) when testing. BTW, I know there is not much literature on the subject.

That said, I was confused because of using the verb `test` in the test name. It's a test so obviously it's testing something. Also because in JUnit 3 you had to add the `test` prefix so the runner knew that was a test and execute it. Not anymore with JUnit4+ and the `@Test` annotation in which adding `test` is not necessary.